src/file/file.repository.ts

```diff
@@ -20,61 +22,141 @@ const sort = {
+  static create(file: any): Promise<IFile> {
     file.parent = file.parent ? new ObjectID(file.parent) : null;
     return fileModel.create(file);
   }
 
   /**
-   * Updates the file metatata by its id.
+  * Add a given shortcut file to the DB.
+  * @param file - is the shortcut file to be added to the DB
+  */
+  static async createShortcut(file: any): Promise<IFile> {
+    const populatedShortcut = await this.baseFileToIFile(await shortcutModel.create(file));
+    const shortcutAsFile: IFile = { ...(populatedShortcut.fileID as any), ...populatedShortcut };
+
+    return shortcutAsFile;
+  }
+
+  /**
+   * Get a baseModel file and returns its model
+   * @param file - the file to get the model from
+   * @returns file's model
+   */
+  static getFileModel(baseFile: any): any {
+    const factoryFile = this.fileFactory(baseFile, baseFile.fileModel);
+    if (factoryFile instanceof IShortcut) {
+      return shortcutModel;
+    }
+    if (factoryFile instanceof IFile) {
+      return fileModel;
+    }
+    return null;
+  }
+
+  /**
+   * Update a file metatata by its id.
    * @param id - the file id.
    * @param partialFile - the partial file containing the attributes to be changed.
    */
   static async updateById(id: string, partialFile: Partial<IFile>): Promise<boolean> {
-    const file = await fileModel.findById(id);
+    const baseFile = await baseFileModel.findById(id);
+    const file = await this.baseFileToIFile(baseFile);
     if (!file) throw new FileNotFoundError();
+    const model = this.getFileModel(baseFile);
+    const res = await model.findByIdAndUpdate(id, { $set: partialFile }, { runValidators: true }).exec();
 
-    const res = await fileModel.updateOne({ _id: file._id, ownerID: file.ownerID }, { $set: partialFile }, { runValidators: true }).exec();
-    // const res = await file.updateOne(partialFile, { runValidators: true }).exec();
-    return res.n === 1 && res.nModified === 1 && res.ok === 1;
+    return res.isModified();
   }
 
   /**
-   * Deletes a file from the DB.
+   * Delete a file from the DB.
    * @param id - the id of the file to be deleted.
    */
   static deleteById(id: string): Promise<IFile | null> {
     return fileModel.findByIdAndRemove({ _id: new ObjectID(id) }).exec();
   }
 
   /**
-   * Get the file by its id.
+   * Get an IPopulatedShortcut type file and converts it to IFile types.
+   * @param file - the file that will be converted.
+   */
+  static populatedShortcutToFile(file: IPopulatedShortcut): IFile {
+    const shortcutAsFile: IFile = { ...file.fileID, ...file };
+    delete shortcutAsFile.fileID;
+    return shortcutAsFile;
+  }
+
+  /**
+   * Get a baseFile type file and converts its type to IFile.
+   * @param file - the file that will be converted.
+   */
+  static async baseFileToIFile(file: any): Promise<IFile> {
+    const factoryFile: PrimitiveFile = this.fileFactory(file, file.fileModel);
+    let responseFile: IFile = factoryFile as IFile;
```

Comment:
Can you explain why factoryFile is already IFile? (in a comment)

Reply:
yes, done.